src/Controller/CancelLastPayPalPaymentAction.php

```diff
@@ -51,6 +51,14 @@ public function __invoke(Request $request): Response
         $paymentStateMachine = $this->stateMachineFactory->get($payment, PaymentTransitions::GRAPH);
         $paymentStateMachine->apply(PaymentTransitions::TRANSITION_CANCEL);
 
+        /** @var PaymentInterface $lastPayment */
+        $lastPayment = $order->getLastPayment();
```

Comment:
Please check for last payment's factory, before using it.

This clause needs to be at https://github.com/Sylius/PayPalPlugin/blob/1.5/src/Processor/OrderPaymentProcessor.php as existing design calls for it.